### PR TITLE
[DRAFT] Evaluate feasability of using FunctionalTensor for Example Value

### PR DESCRIPTION
**Pull request by Lucaskabela**
WIP - gathering initial signals 

To replace one view and mutation metadata pass

Key test:

These 4 cases we were most concerned with can be ran with 
```
python test.py
```
```
python test/functorch/test_aotdispatch.py 
```
Results in
```
----------------------------------------------------------------------
Ran 2131 tests in 2267.300s

FAILED (failures=53, errors=49, skipped=450, expected failures=60)
```

Sample failure:
```
    python test/functorch/test_aotdispatch.py TestAOTExport.test_aot_export_with_torch_cond
```
Like
```
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/convert_frame.py", line 784, in compile_inner
    return _compile_inner(code, one_graph, hooks, transform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/convert_frame.py", line 823, in _compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/bytecode_transformation.py", line 1424, in transform_code_object
    transformations(instructions, code_options)
  File "/home/lucaskabela/pytorch/torch/_dynamo/convert_frame.py", line 266, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/convert_frame.py", line 744, in transform
    tracer.run()
  File "/home/lucaskabela/pytorch/torch/_dynamo/symbolic_convert.py", line 3488, in run
    super().run()
  File "/home/lucaskabela/pytorch/torch/_dynamo/symbolic_convert.py", line 1359, in run
    while self.step():
          ^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/symbolic_convert.py", line 1263, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/lucaskabela/pytorch/torch/_dynamo/symbolic_convert.py", line 831, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/symbolic_convert.py", line 2221, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars)
  File "/home/lucaskabela/pytorch/torch/_dynamo/symbolic_convert.py", line 1189, in call_function
    self.push(fn.call_function(self, args, kwargs))  # type: ignore[arg-type]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/variables/lazy.py", line 201, in realize_and_forward
    return getattr(self.realize(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 75, in graph_break_as_hard_error
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 1086, in call_function
    (true_r, true_treespec, true_graph, true_lifted_freevars) = speculate_branch(
                                                                ^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 1061, in speculate_branch
    ) = speculate_subgraph(
        ^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 823, in speculate_subgraph
    aliasing_info = subtracer.has_aliasing()
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_dynamo/output_graph.py", line 3065, in has_aliasing
    example_value = _collect_fake_inputs([node])[0]
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucaskabela/pytorch/torch/_higher_order_ops/utils.py", line 372, in _collect_fake_inputs
    assert isinstance(val, FakeTensor)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

File: torch/_dynamo/utils.py [hunk 1]
```diff
@@ -3245,7 +3246,7 @@ def get_fake_value(node, tx, allow_non_graph_fake=False):
         )
 
     try:
-        with tx.fake_mode, enable_python_dispatcher():
+        with tx.functional_mode, tx.fake_mode, enable_python_dispatcher():
```
Comment (Lucaskabela) [inline torch/_dynamo/utils.py:3288]: run_node has issues here for `self.bn` - namely, several lines above we perform a deep copy on the nnmodule for BatchNorm (https://github.com/pytorch/pytorch/blob/12b02137af5cd6743adfe5a0ffe0d9b453cd013e/torch/_dynamo/utils.py#L3234)

There are seemingly two conflicting things here:
1. the `example_value` in node extracted here needs to be functional to record potential mutations when running the node, and a functional tensor can only be acted on by other functional tensors
2. the `deep copy` linked above does not support functional tensors for the Parameter's (such as running_mean) as the torch__dispatch__ does not support Parameter types

So the seeming solution would be to enable deep copy on Parameter type for functional tensors, but since these are parameters aren't they inheritly stateful?
